tooling/miniutil/src/build/terminator.rs

```diff
@@ -1,5 +1,253 @@
 use crate::build::*;
 
+impl FunctionBuilder {
+    #[track_caller]
+    fn finish_block(&mut self, terminator: Terminator) {
+        let bb = BasicBlock { statements: self.cur_block().statements, terminator };
```

Comment:
`finish_block` isn't the only function using `cur_block()`, the statements in `statements.rs` are using it as well. 
The approach with `take` would allow access to the block in two different ways.

Reply:
> finish_block isn't the only function using cur_block(), the statements in statements.rs are using it as well.

Ah, makes sense.

I think using `take` here would still be more idiomatic Rust though.